### PR TITLE
WebDriver conformance changes for Firefox 87

### PR DESCRIPTION
**Pull request by whimboo**
CC'ing @juliandescottes for info.

### CODE REVIEW

File: files/en-us/mozilla/firefox/releases/87/index.html [hunk 1]
```diff
@@ -92,7 +92,15 @@ <h4 id="DOM">DOM</h4>
...
 
-<p><em>No changes.</em></p>
+<ul>
+  <li>With the Fission (site-isolation) support finished, all the old code has been removed. As such the preference <code>marionette.actors.enabled</code> is no longer available ({{bug(1669172)}}).</li>
+  <li>WebDriver commands following a call to <code>WebDriver:SwitchToFrame</code> will no longer fail with a "no such window" error if the frame's content hasn't yet finished loading ({{bug(1691348)}}).</li>
+  <li>After a <a
+      href="https://firefox-source-docs.mozilla.org/dom/navigation/nav_replace.html#cross-group-navigations">cross-group
+      page navigation</a>, accessing a previously-retrieved element will now always raise a "stale element" error, and
+    no longer might lead to a "no such element" error ({{bug(1690308)}}).</li>
+  <li><code>Addon:Uninstall</code> now raises an <code>unknown error</code> when the id of the add-on to uninstall is unknown ({{bug(1690308)}}).</li>
```
Comment (juliandescottes) [inline files/en-us/mozilla/firefox/releases/87/index.html:102]: The bug was https://bugzilla.mozilla.org/show_bug.cgi?id=1693022
```suggestion
  <li><code>Addon:Uninstall</code> now raises an <code>unknown error</code> when the id of the add-on to uninstall is unknown ({{bug(1693022)}}).</li>
```

File: files/en-us/mozilla/firefox/releases/87/index.html [hunk 2]
```diff
@@ -92,7 +92,15 @@ <h4 id="DOM">DOM</h4>
 
 <h3 id="webdriver_conformance_marionette">WebDriver conformance (Marionette)</h3>
 
-<p><em>No changes.</em></p>
+<ul>
+  <li>With the Fission (site-isolation) support finished, all the old code has been removed. As such the preference <code>marionette.actors.enabled</code> is no longer available ({{bug(1669172)}}).</li>
```
Comment (chrisdavidmills) [inline files/en-us/mozilla/firefox/releases/87/index.html:96]: I don't really understand this entry. Do you mean to say that Fission (site-isolation) support is now enabled by default, therefore some kind of related older code has been removed as it is no longer relevant, which includes a related preference? The way that you've used "finished" could also imply that fission support has ended, i.e. is no longer available. 

It would be good to try and rephrase this to clarify. Thanks!

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Thanks for the addition @whimboo ! I did a brief copy edit, and also raised a question about a part of the text that I didn't understand.

File: files/en-us/mozilla/firefox/releases/87/index.html (hunk 2 above)
Comment (whimboo) [reply files/en-us/mozilla/firefox/releases/87/index.html:96]: Thanks for the note. So the work we had to do to support Fission has been finished. As such the code that was run before (message managers) has been completely deleted. Also we enabled the Fission code already 2 releases ago by default, but kept this preference around for a possible fallback when something could have been totally broken.  How would you write it instead?

File: files/en-us/mozilla/firefox/releases/87/index.html (hunk 1 above)
Comment (whimboo) [reply files/en-us/mozilla/firefox/releases/87/index.html:101]: Ups. Bad copy and paste. Thanks.

**Comment by whimboo** [comment]
@chrisdavidmills sorry I didn't notice that you already made a copy edit. Maybe it was the same what @juliandescottes mentioned? I did a force push and as such removed your change. :/

File: files/en-us/mozilla/firefox/releases/87/index.html (hunk 2 above)
Comment (whimboo) [reply files/en-us/mozilla/firefox/releases/87/index.html]: I did a small edit here. Let me know if that's better. Otherwise maybe you can directly do the edit?

**Comment by chrisdavidmills** [comment]
> @chrisdavidmills sorry I didn't notice that you already made a copy edit. Maybe it was the same what @juliandescottes mentioned? I did a force push and as such removed your change. :/

No worries. It was an edit on the third bullet to improve some awkward wording. I've added it back in now.

File: files/en-us/mozilla/firefox/releases/87/index.html (hunk 2 above)
Comment (chrisdavidmills) [reply files/en-us/mozilla/firefox/releases/87/index.html]: OK, I think I'm understanding this better now.

Does the following sound OK?

"Since Fission (site-isolation) is now finalized and enabled by default, we no longer need to maintain the fallback message manager code that was required to run WebDriver in pre-fission Firefox. As a result, this code has been deleted, and we have also removed the <code>marionette.actors.enabled</code> preference that enabled it ({{bug(1669172)}})."

Comment (whimboo) [reply files/en-us/mozilla/firefox/releases/87/index.html]: Here with some small adjustments:

"Since support for Fission (site-isolation) is now finalized and got enabled by default with Firefox 84, we no longer need to maintain the fallback message manager code that was used to run Marionette in pre-Fission Firefox. As a result, this code has been deleted, and we have also removed the `marionette.actors.enabled` preference that was used to turn on/off this feature ({{bug(1669172)}})."

Comment (juliandescottes) [reply files/en-us/mozilla/firefox/releases/87/index.html]: > Since Fission (site-isolation) is now finalized and enabled by default

Slightly worried this could be interpreted as if the overall Fission work in Firefox is finalized and enabled, which is not the case yet. 

When we say "Fission code" here, we actually mean the Marionette rewrite to support Fission. This rewrite works whether Fission is enabled or not. But we kept the old implementation available behind a preference as a safety measure. It also allowed us to quickly know if an issue was linked to the rewrite because we could ask bug reporters to try their scenarios with the preference set to false. We feel like we have had enough time to collect and fix regressions now, so it's safe to always use the rewrite.

**Comment by whimboo** [comment]
I just updated. @juliandescottes is it better now? Not sure if we should go into too much detail here.

**Comment by chrisdavidmills** [comment]
OK, *now* I understand ;-)

How about this rewrite?

"The work of rewriting Marionette to work with Fission (site-isolation) has been finished, so the old Marionette implementation has been removed. The <code>marionette.actors.enabled</code> preference, which toggled between the new and old implementations, has therefore also been removed ({{bug(1669172)}})."

**Comment by whimboo** [comment]
> The work of rewriting Marionette to work with Fission

Maybe just a small change... `The work of rewriting Marionette to support Fission`. That way we don't have the word `work` twice after each other.

**Comment by chrisdavidmills** [comment]
> > The work of rewriting Marionette to work with Fission
> 
> Maybe just a small change... `The work of rewriting Marionette to support Fission`. That way we don't have the word `work` twice after each other.

Yup, sounds great!

File: files/en-us/mozilla/firefox/releases/87/index.html (hunk 2 above)
Comment (whimboo) [reply files/en-us/mozilla/firefox/releases/87/index.html]: Everyone happy now? Just pushed an update for the recent suggestion.

**Review by chrisdavidmills — APPROVED** [review]
Yup, I'm happy! Any other comments @juliandescottes ?

**Comment by juliandescottes** [comment]
LGTM!
